src/radar/radar/quantum.py

```diff
@@ -1,58 +1,166 @@
-    
-    # Create command socket
-    with socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM, proto=socket.IPPROTO_UDP) as command_socket:
-        command_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        command_socket.setblocking(0) # non-blocking socket
-        logger.info(f'Initialized command_socket')
+from radar.packets.location_info import LocationInfo
+from radar.packets import control_message
+
+
+class Qauntum(Node):
+
+    def __init__(self):
+        super().__init__('quantum')
+        # uncomment below to simulate locator
+        # data = bytes.fromhex('00000000928b80cb28000000030064000608100001b301e80e0a11007501a8c00f0a3600')
+        # bl = LocationInfo.parse(data[:36])
+        # self.quantum_location = LocationInfo(*bl)
+        self.quantum_location = self.locate_quantum() # this might take a while to return
         
-        logger.info(f'Run radar keep alive thread')
-        keep_alive_thread = Thread(target = radar.control.standby_radar, args=(command_socket, li.radar_ip, li.radar_port))
-        keep_alive_thread.start()
+        self.command_socket = self.create_command_socket() # socket for controlling radar
+        self.report_socket = self.create_report_socket() # socket for receiving radar data
         
-        radar.control.radar_tx_on(command_socket=command_socket,
-                                  command_ip=li.radar_ip,
-                                  command_port=li.radar_port)
+        # keey radar alive
+        self.get_logger().info(f'Keeping radar alive')
+        self.standby_timer = self.create_timer(1, self.standby_timer_callback) # 1 second timer
+        self.alive_counter = 0
         
-        # Create report socket
-        with socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM, proto=socket.IPPROTO_UDP) as report_socket:
-            report_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-            
-            # Bind to the server address
-            # on this port, receives ALL multicast groups
-            report_socket.bind(('', li.data_port))
+        # subscription
+        self.radar_control_subscription = self.create_subscription(
+            String,
+            'radar_control',
+            self.radar_control_callback,
+            10)
+        
+        # service
+        self.radar_data_srv = self.create_service(
+            RadarData, 
+            'radar_data', 
+            self.radar_data_callback)
+
+
+    def locate_quantum(self, multicast_group='224.0.0.1', multicast_port=5800, quantum_model_id=40) -> LocationInfo:
+        # Create UDP socket
+        locator_socket = socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM, proto=socket.IPPROTO_UDP)
+        locator_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        
+        # Bind to the server address
+        # on this port, receives ALL multicast groups
+        locator_socket.bind(('', multicast_port))
```

Comment:
You might need to open the listener on a specific interface rather than letting the OS decide which one. Check with `ss`

Reply:
This one's the locator socket that listens on 224.0.0.1:5800 that works reliably. The problematic one is the report socket that listens on 232.1.179.1:2574.